### PR TITLE
feat(lib): convert question utility to TypeScript

### PR DESCRIPTION
**Pull request by pinssbrett**
## Describe your changes
Convert `util/questions.js` to TypeScript. Remove some tests to reflect narrower typing of function parameters, and add some new tests to reflect boundary conditions.

## Issue ticket number and link
N/A

### CODE REVIEW

File: packages/lib/util/questions.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+export function referenceDataToRadioOptionsWithHintText(
+	reference: ReferenceWithHint[],
+	addNullOption: boolean = false
+): Option[] {
+	const options = reference.map((t) => ({
+		text: t.displayName,
+		value: t.id,
+		hint: { text: t.hintText }
+	}));
```
[the remark this answers is not in the document]
Comment (pinssbrett) [reply packages/lib/util/questions.ts:35]: New narrow types mean that these values will always be set

File: packages/lib/util/questions.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		return `${truncated}... `;
+	}
+	return comment;
+}
+
+/**
+ * Generates a "Read more" link for a truncated comment, directing users to the full comment.
+ */
+export function truncatedReadMoreCommentLink(href: string): string {
+	return `<a class="govuk-link govuk-link--no-visited-state" href="${href}">Read more</a>`;
```
[the remark this answers is not in the document]
Comment (pinssbrett) [reply packages/lib/util/questions.ts:74]: Out of scope: to handle when converting call site to TS

File: packages/lib/util/questions.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const MAX_LENGTH = 500;
+
+type Reference = {
+	id: string;
+	displayName: string;
+};
+
+type ReferenceWithHint = Reference & { hintText: string };
+
```
[the remark this answers is not in the document]
Comment (pinssbrett) [reply packages/lib/util/questions.ts:12]: Intentionally strict to reflect current behaviour - add in checks downstream when converted to TS if necessary

File: packages/lib/util/questions.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+/**
+ * Truncates a comment if it exceeds the maximum length and appends an ellipsis.
+ */
+export function truncateComment(comment: string): string {
+	if (shouldTruncateComment(comment)) {
+		const truncated = comment.substring(0, MAX_LENGTH);
+		return `${truncated}... `;
+	}
+	return comment;
+}
```
Comment (oscargay) [inline packages/lib/util/questions.ts:68]: Out of scope for this PR, just mentioning, but always thought that these `util`s could sit in a more generic `strings.ts` util, and then have MAX_LENGTH as a second, defaulted parameter, as they're generic enough as is and aren't tied to just the questions module.

e.g. `export function shouldTruncateComment(comment: string, maxLength = MAX_LENGTH): boolean {`

**Review by oscargay — APPROVED** [review]
LGTM, left a minor comment that is out of scope for this conversion PR but just crossed my mind when reading it

File: packages/lib/util/questions.ts (hunk 4 above)
Comment (pinssbrett) [reply packages/lib/util/questions.ts:68]: Thanks, will add to backlog!
